### ISSUE TITLE
destructBoxes is not destructing all the boxes

### ISSUE THREAD

**Issue by MareoRaft** [abrobbins90/groceries#3]
When adding things to the recipe area, destructBoxes does not destroy all the boxes, so that there then appears redundant boxes.

Not sure why.  pulling hair out.

**Comment by MareoRaft** [comment]
@abrobbins90 

Also, I think it's weird that Closet's method "remove" doesn't actually destruct the Box.  I feel like I got it backwards.  I think Closet's .remove should be called .destructBox, and it should call the box's .destruct, or something, instead of the other way around.

I still couldn't pin down the error on this one :(

I had an idea for how to make Closet a little smarter which would probably make you happy.  Closet could have a method called `repopulate nodes` that takes in a list of nodes.  Instead of emptying the whole Closet and rebuilding from scratch, the closet would...

1. See which boxes it currently has that are *not* in the input list of nodes.  Destruct those.
2. See which nodes in the input list of nodes do not have a box in the closet already.  Construct those.

I'm in favor of making some sort of change, because it might fix the bug in the process :)

Matt

**Comment by abrobbins90** [comment]
Fixed the problem. I vaguely thought this could be a problem when we programmed it yesterday. The destruct boxes method in the closet loops over this.boxes. However each iteration then deletes an element from this.boxes. Thus every other box was deleted. I duplicated the variable so it's not changed in the loop. The other fix would be to have it iterate backwards. But this way works fine.

I'll begin implementing these things you mentioned. I too when going through found the whole removal process a bit convoluted.
